ruma-federation-api/src/event/get_room_state_ids/v1.rs

```diff
@@ -0,0 +1,52 @@
+//! [GET /_matrix/federation/v1/state_ids/{roomId}](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-state-ids-roomid)
+
+use ruma_api::ruma_api;
+use ruma_identifiers::{EventId, RoomId};
+
+ruma_api! {
+    metadata: {
+        description: "Retrieves a snapshot of a room's state at a given event, in the form of event IDs",
+        method: GET,
+        name: "get_room_state_ids",
+        path: "/_matrix/federation/v1/state_ids/:room_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    request: {
+        /// The room ID to get state for.
+        #[ruma_api(path)]
+        room_id: &'a RoomId,
+
+        /// An event ID in the room to retrieve the state at.
+        #[ruma_api(query)]
+        event_id: &'a EventId,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    response: {
+        /// The full set of authorization events that make up the state of the
+        /// room, and their authorization events, recursively.
+        auth_chain_ids: Vec<EventId>,
+
+
+        /// The fully resolved state of the room at the given event.
+        pdu_ids: Vec<EventId>
```

Comment:
Not sure if something besides event id formatted data can be returned here, spec just says string.

Reply:
Think this is actually fine

Reply:
Sorry I'm not used to github's reviewing stuff.

for reference what was deleted was (paraphrasing)

> As far as the state resolution side of things goes, event ids are the only id type used.